v1/tarball/image.go

```diff
@@ -177,6 +182,89 @@ func (i *image) RawConfigFile() ([]byte, error) {
 	return i.config, nil
 }
 
+// Flatten takes an image and flattens its layers into a single tarball at the specified path
+func Flatten(img v1.Image, tarFilePath string) error {
+	tarFile, err := os.Create(tarFilePath)
+	if err != nil {
+		return fmt.Errorf("Error creating tar file: %v", err)
+	}
+	defer tarFile.Close()
+	fileWriter := bufio.NewWriter(tarFile)
+	defer fileWriter.Flush()
+	tarWriter := tar.NewWriter(fileWriter)
+	defer tarWriter.Close()
+
+	fileMap := make(map[string]bool, 0)
+
+	layers, err := img.Layers()
+	if err != nil {
+		return fmt.Errorf("Error retriving image layers: %v", err)
+	}
+	for i := len(layers) - 1; i >= 0; i-- {
```

Comment:
Can you add a comment on why we go in reverse?

An overall link to the python implementation might help people understand.

Reply:
done